test/call.c

```diff
@@ -197,72 +196,141 @@ static struct cancel_rule *fixture_add_cancel_rule(struct fixture *f,
 	r->n_progress    = n_progress;
 	r->n_established = n_established;
 
-	list_append(&f->rules, &r->le, r);
+	r->n_audio_estab = (unsigned) -1;
+	r->n_video_estab = (unsigned) -1;
+	r->n_offer_cnt   = (unsigned) -1;
+	r->n_answer_cnt  = (unsigned) -1;
+	r->n_vidframe    = (unsigned) -1;
```

Comment:
Maybe better use `int` as type?

Reply:
ah, yes.

Currently I am confused why the SIP 183 is coming twice in valgrind retest check. Maybe the SIP retry mechanism causes this?

Not sure if we should check `n_progress==1` or `n_progress>=1`.

Reply:
Today I already found an issue because of the `==` check. The `call_progress()` was called twice. So I could fix this by
dropping `BEHAVIOUR_PROGRESS`. So `==` check might be useful in general.